content/en/articles/js/flow/index.md

```diff
@@ -0,0 +1,290 @@
+---
+date: "2016-06-07"
+title: Introduction to Flow, to make your code run on the first try
+tags:
+  - javascript
+  - flow
+  - type
+authors:
+  - bloodyowl
+  - skinnyfoetusboy
+---
+
+```javascript
+document.body.firstChild.getBoundingClientRect()
+```
+
+This hypothetic line of code doesn't always work.
+
+The reason for this is that `element.firstChild` is a `Node` and that not all `Nodes` have a `getBoundingClientRect` method, which is always found on `Elements`, a class that inherits from `Node`.
```

Comment:
...is a `Node` not all of which have a `getBoundingClientRect` method, but which is always found...